tests/objects/toop1.nim

```diff
@@ -34,25 +34,30 @@ proc init(my: var TRectangle) =
   my.height = 10
   my.draw = cast[proc (my: var TFigure) {.nimcall.}](drawRectangle)
 
-macro `!` (n: varargs[untyped]): typed =
-  let n = callsite()
-  result = newNimNode(nnkCall, n)
-  var dot = newNimNode(nnkDotExpr, n)
-  dot.add(n[1])    # obj
-  if n[2].kind == nnkCall:
+macro `!` (args: varargs[untyped]): typed =
+  args.expectLen 2
+  let arg1 = args[0]
+  let arg2 = args[1]
```

Comment:
Why alias instead of a simple `n[1]` -> `args[0]` replacement?

Reply:
actually I think it would be best to have arg1 and arg2 directly in the argument list to the macro. For some reasons that I don't really want to dig in, I didn't work out. The test is really old and has many more problems than this. I didn't plan to really refactor this test.